nix/nixpkgs-with-haskell-nix.nix

```diff
@@ -1,10 +1,20 @@
 # SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
 # SPDX-License-Identifier: MPL-2.0
-
 let
   sources = import ./sources.nix;
-  haskellNix = import sources."haskell.nix" {
-    sourceOverrides = { hackage = sources."hackage.nix"; stackage = sources."stackage.nix"; };
-  };
+  haskellNix = import sources."haskell.nix" {};
   nixpkgs = import sources.nixpkgs;
-in nixpkgs haskellNix.nixpkgsArgs
+
+  # override hackage and stackage index pins to update them independently of haskell.nix
+  pinHackageOverlay = self: super: {
+    haskell-nix = super.haskell-nix // {
+      hackageSrc = sources."hackage.nix";
+      stackageSrc = sources."stackage.nix";
+    };
+  };
+
+  nixpkgsArgs = haskellNix.nixpkgsArgs // {
+    overlays = haskellNix.nixpkgsArgs.overlays ++ [ pinHackageOverlay ];
+  };
+
+in nixpkgs nixpkgsArgs
```

Comment:
> Is this problem present in other repos?

No, I don't think so. AFAIU, the problem was caused by `sourceOverrides`, which was used instead of adding an overlay with pinned stackage and hackage.

Reply:
Why do we need this file at all? I don't see such file in `tezos-btg` and `tezos-btc`, for example.

Reply:
because haskell.nix by default uses some fixed revisions of hackage.nix and stackage.nix, thus we have to updated haskell.nix itself when we want to update hackage or stackage, thus we very highly likely to rebuild ghc in CI

Reply:
and `tezos-btg` along with `tezos-btc` don't use stable morley and lorentz, thus they don't need fresh hackage

Reply:
Why can't we do it like in https://github.com/serokell/tezos-btg/pull/71/files?

Reply:
Looks like there was a typo, it should be `sourcesOverride` instead, see https://serokell.slack.com/archives/C742ZKYF5/p1599461917083500?thread_ts=1599231595.057200&cid=C742ZKYF5. Also, looks like it indeed makes more sense to inline this file to the `ci.nix`

Reply:
@gromakovsky ku-ku

Reply:
zdaroff

Reply:
Alright, good job!